File: crates/test-support/src/integration_tests/config/negative_refspec.rs
use crate::integration_test::{IntegrationTest, NewIntegrationTestArgs};

pub struct NegativeRefspecTest;

impl NegativeRefspecTest {
    pub fn new() -> IntegrationTest {
        IntegrationTest::new(NewIntegrationTestArgs {
            description: "Having a config with a negative refspec".to_string(),
            extra_cmd_args: vec![],
            skip: false,
            setup_repo: Some(Box::new(|_shell| {})),
            setup_config: Some(Box::new(|_cfg| {})),
            run: Some(Box::new(|_t, _keys| {})),
            extra_env_vars: std::collections::HashMap::new(),
            git_version: Default::default(),
            width: 0,
            height: 0,
            is_demo: false,
        })
    }
}

pub const NEGATIVE_REFSPEC: NegativeRefspecTest = NegativeRefspecTest;

File: pkg/integration/tests/config/negative_refspec.go
package config

import (
	"github.com/quangdang46/slg/pkg/config"
	. "github.com/quangdang46/slg/pkg/integration/components"
)

var NegativeRefspec = NewIntegrationTest(NewIntegrationTestArgs{
	Description:  "Having a config with a negative refspec",
	ExtraCmdArgs: []string{},
	SetupRepo: func(shell *Shell) {
		shell.
			SetConfig("remote.origin.fetch", "^refs/heads/test").
			CreateNCommits(2)
	},
	SetupConfig: func(cfg *config.AppConfig) {},
	Run: func(t *TestDriver, keys config.KeybindingConfig) {
		// the failure case with an unpatched go-git is that no branches display
		t.Views().Branches().
			Lines(
				Contains("master"),
			)
	},
})
